### DISCUSSION TITLE
how to dynamic adjust speed?

### DISCUSSION THREAD

**Discussion by Magic-chao** [mjpieters/aiolimiter#104] [Q&A]
my first step：
limiter = AsyncLimiter(1,1/0.5)

after 5s，I want to set qps=1，but it doesn't work

limiter.time_period = 1

**Comment by mjpieters** [comment]
This isn't something that is supported sorry.

The `time_period` value is used to calculate a private rate/sec value and _that_ value is then used to handle the bucket capacity. You would have to set `limiter._rate_per_sec`, but a future version of aiolimiter may use a different parameter name. The risk is yours.

Why not, instead, replace the limiter whenever you need to adjust the rate?
